### PR TITLE
ENH: Named aggregation in SeriesGroupBy.agg

### LINKED ISSUE #26512: Allow Keyword aggregation in SeriesGroupBy
Followup to https://github.com/pandas-dev/pandas/pull/26399

```python
In [1]: df = pd.DataFrame({'A': [1, 1, 1, 2, 2], 
   ...:                      'B': range(5), 
   ...:                      'C': range(5)})                                    

In [2]: df                                     
Out[2]: 
   A  B  C
0  1  0  0
1  1  1  1
2  1  2  2
3  2  3  3
4  2  4  4

In [3]: df.groupby('A').B.agg(foo='count')                                                        
Out[3]: 
   foo
A     
1    3
2    2
```

https://github.com/pandas-dev/pandas/pull/26399#pullrequestreview-241906201

### PR DESCRIPTION
**Pull request by TomAugspurger**
```python
In [4]:    animals = pd.DataFrame({'kind': ['cat', 'dog', 'cat', 'dog'],
   ...:                            'height': [9.1, 6.0, 9.5, 34.0],
   ...:                            'weight': [7.9, 7.5, 9.9, 198.0]})
   ...:    animals.groupby("kind").height.agg(max_height='max')
Out[4]:
      max_height
kind
cat          9.5
dog         34.0
```

Closes #26512

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -52,12 +52,26 @@ what the arguments to the function are, but plain tuples are accepted as well.
...
+.. ipython:: python
+
+   animals.groupby("kind").height.agg(
+       min_height="min",
+       max_height="max",
+   )
+
+
+This type of aggregation is the recommended alternative to the deprecated behavior when passing
+a dict to a Series groupby aggregation (:ref:`whatsnew_0200.api_breaking.deprecate_group_agg_dict`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: don't you need
``:ref:`see here<whatsnew.....>` `` for this to render (and the above one)?

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 2]
```diff
@@ -331,6 +331,37 @@ def test_uint64_type_handling(dtype, how):
 
 class TestNamedAggregation:
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py]: could make this a separate class

**Review by jreback — CHANGES_REQUESTED** [review]
small comments

**Review by WillAyd — APPROVED** [review]
lgtm ex @jreback  comments

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.25.0.rst]: It takes the title of the section by default.

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 2 above)
Comment (TomAugspurger) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: You're saying split the series specific ones, into TestNamedAggregationSeries and TestNamedAggregationDataFrame?

Comment (jreback) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: yes

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (jreback) [reply doc/source/whatsnew/v0.25.0.rst]: ok

**Comment by jreback** [comment]
can you merge master to ensure passing, otherwise lgtm.

**Comment by TomAugspurger** [comment]
All green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**